src/java.base/unix/classes/java/lang/ProcessImpl.java

```diff
@@ -92,65 +90,40 @@ private static enum LaunchMechanism {
         VFORK
     }
 
-    private static enum Platform {
-
-        LINUX(LaunchMechanism.POSIX_SPAWN, LaunchMechanism.VFORK, LaunchMechanism.FORK),
-
-        BSD(LaunchMechanism.POSIX_SPAWN, LaunchMechanism.FORK),
-
-        AIX(LaunchMechanism.POSIX_SPAWN, LaunchMechanism.FORK);
-
-        final LaunchMechanism defaultLaunchMechanism;
-        final Set<LaunchMechanism> validLaunchMechanisms;
-
-        Platform(LaunchMechanism ... launchMechanisms) {
-            this.defaultLaunchMechanism = launchMechanisms[0];
-            this.validLaunchMechanisms =
-                EnumSet.copyOf(Arrays.asList(launchMechanisms));
+    /**
+     * {@return the default or requested launch mechanism}
+     * @throw Error if the requested launch mechanism is not found or valid
+     */
+    @SuppressWarnings("removal")
+    private static LaunchMechanism launchMechanism() {
+        String s = GetPropertyAction.privilegedGetProperty("jdk.lang.Process.launchMechanism");
+        if (s == null) {
+            return LaunchMechanism.POSIX_SPAWN;
         }
 
-        @SuppressWarnings("removal")
-        LaunchMechanism launchMechanism() {
-            return AccessController.doPrivileged(
-                (PrivilegedAction<LaunchMechanism>) () -> {
-                    String s = System.getProperty(
-                        "jdk.lang.Process.launchMechanism");
-                    LaunchMechanism lm;
-                    if (s == null) {
-                        lm = defaultLaunchMechanism;
-                        s = lm.name().toLowerCase(Locale.ROOT);
-                    } else {
-                        try {
-                            lm = LaunchMechanism.valueOf(
-                                s.toUpperCase(Locale.ROOT));
-                        } catch (IllegalArgumentException e) {
-                            lm = null;
-                        }
-                    }
-                    if (lm == null || !validLaunchMechanisms.contains(lm)) {
-                        throw new Error(
-                            s + " is not a supported " +
-                            "process launch mechanism on this platform."
-                        );
+        try {
+            // Should be value of a LaunchMechanism enum
+            LaunchMechanism lm = LaunchMechanism.valueOf(s.toUpperCase(Locale.ENGLISH));
```

Comment:
I think `Locale.ROOT` is preferred here.

Reply:
ok, but not strictly in scope for this PR; that's pre-existing code.